Replace c10::guts::stuff with std::stuff

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30924 Reimplement c10::apply / c10::invoke with c++14
* #30922 Simplify code with decltype(auto)
* #31091 if_constexpr for C++14
* **#30915 Replace c10::guts::stuff with std::stuff**

Since we now have C++14, we don't need these c10::guts helpers anymore

- c10::guts::make_unique -> std::make_unique
- c10::guts::integer_sequence -> std::integer_sequence
- c10::guts::index_sequence -> std::index_sequence
- c10::guts::make_integer_sequence -> std::make_integer_sequence
- c10::guts::make_index_sequence -> std::make_index_sequence
- c10::guts::index_sequence_for -> std::index_sequence_for
- c10::guts::conditional_t -> std::conditional_t
- c10::guts::enable_if_t -> std::enable_if_t
- c10::guts::add_lvalue_reference_t -> std::add_lvalue_reference_t
- c10::guts::remove_reference_t -> std::remove_reference_t
- c10::guts::remove_cv_t -> std::remove_cv_t
- c10::guts::result_of_t -> std::result_of_t
- c10::guts::decay_t -> std::decay_t
- c10::guts::remove_const_t -> std::remove_const_t
- c10::guts::remove_pointer_t -> std::remove_pointer_t
- c10::guts::common_type_t -> std::common_type_t

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

smessmer committed:
Replace c10::guts::stuff with std::stuff

Since we now have C++14, we don't need these c10::guts helpers anymore

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

[ghstack-poisoned]

ezyang reviewed (approved):
As a matter of the pubic record can you list in the PR description all of the functions you converted.;

smessmer committed:
Update on "Replace c10::guts::stuff with std::stuff"


Since we now have C++14, we don't need these c10::guts helpers anymore

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

[ghstack-poisoned]

smessmer committed:
Update on "Replace c10::guts::stuff with std::stuff"


Since we now have C++14, we don't need these c10::guts helpers anymore

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

[ghstack-poisoned]

smessmer committed:
Update on "Replace c10::guts::stuff with std::stuff"


Since we now have C++14, we don't need these c10::guts helpers anymore

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

[ghstack-poisoned]

smessmer committed:
Update on "Replace c10::guts::stuff with std::stuff"


Since we now have C++14, we don't need these c10::guts helpers anymore

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

[ghstack-poisoned]

smessmer committed:
Update on "Replace c10::guts::stuff with std::stuff"


Since we now have C++14, we don't need these c10::guts helpers anymore

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

[ghstack-poisoned]

smessmer committed:
Update on "Replace c10::guts::stuff with std::stuff"


Since we now have C++14, we don't need these c10::guts helpers anymore

- c10::guts::make_unique -> std::make_unique
- c10::guts::integer_sequence -> std::integer_sequence
- c10::guts::index_sequence -> std::index_sequence
- c10::guts::make_integer_sequence -> std::make_integer_sequence
- c10::guts::make_index_sequence -> std::make_index_sequence
- c10::guts::index_sequence_for -> std::index_sequence_for
- c10::guts::conditional_t -> std::conditional_t
- c10::guts::enable_if_t -> std::enable_if_t
- c10::guts::add_lvalue_reference_t -> std::add_lvalue_reference_t
- c10::guts::remove_reference_t -> std::remove_reference_t
- c10::guts::remove_cv_t -> std::remove_cv_t
- c10::guts::result_of_t -> std::result_of_t
- c10::guts::decay_t -> std::decay_t
- c10::guts::remove_const_t -> std::remove_const_t
- c10::guts::remove_pointer_t -> std::remove_pointer_t
- c10::guts::common_type_t -> std::common_type_t

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

[ghstack-poisoned]

smessmer committed:
Update on "Replace c10::guts::stuff with std::stuff"


Since we now have C++14, we don't need these c10::guts helpers anymore

- c10::guts::make_unique -> std::make_unique
- c10::guts::integer_sequence -> std::integer_sequence
- c10::guts::index_sequence -> std::index_sequence
- c10::guts::make_integer_sequence -> std::make_integer_sequence
- c10::guts::make_index_sequence -> std::make_index_sequence
- c10::guts::index_sequence_for -> std::index_sequence_for
- c10::guts::conditional_t -> std::conditional_t
- c10::guts::enable_if_t -> std::enable_if_t
- c10::guts::add_lvalue_reference_t -> std::add_lvalue_reference_t
- c10::guts::remove_reference_t -> std::remove_reference_t
- c10::guts::remove_cv_t -> std::remove_cv_t
- c10::guts::result_of_t -> std::result_of_t
- c10::guts::decay_t -> std::decay_t
- c10::guts::remove_const_t -> std::remove_const_t
- c10::guts::remove_pointer_t -> std::remove_pointer_t
- c10::guts::common_type_t -> std::common_type_t

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 90374de:
* **1/1** broken upstream at merge base 409151e (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=409151e1bb90c66546901d372f59930f72242ee9&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=90374de77c39eca93c70a353a4b3f872428971d4).

### 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [caffe2_onnx_py2_gcc5_ubuntu16_04_test](https://dr.pytorch.org/build-details.html?build_id=75034609)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

smessmer committed:
Update on "Replace c10::guts::stuff with std::stuff"


Since we now have C++14, we don't need these c10::guts helpers anymore

- c10::guts::make_unique -> std::make_unique
- c10::guts::integer_sequence -> std::integer_sequence
- c10::guts::index_sequence -> std::index_sequence
- c10::guts::make_integer_sequence -> std::make_integer_sequence
- c10::guts::make_index_sequence -> std::make_index_sequence
- c10::guts::index_sequence_for -> std::index_sequence_for
- c10::guts::conditional_t -> std::conditional_t
- c10::guts::enable_if_t -> std::enable_if_t
- c10::guts::add_lvalue_reference_t -> std::add_lvalue_reference_t
- c10::guts::remove_reference_t -> std::remove_reference_t
- c10::guts::remove_cv_t -> std::remove_cv_t
- c10::guts::result_of_t -> std::result_of_t
- c10::guts::decay_t -> std::decay_t
- c10::guts::remove_const_t -> std::remove_const_t
- c10::guts::remove_pointer_t -> std::remove_pointer_t
- c10::guts::common_type_t -> std::common_type_t

Differential Revision: [D18869639](https://our.internmc.facebook.com/intern/diff/D18869639/)

[ghstack-poisoned]
